app/javascript/mastodon/containers/identity_context.tsx

```diff
@@ -0,0 +1,33 @@
+import { createContext, useContext } from 'react';
+
+import type { InitialState } from 'mastodon/initial_state';
+
+interface IdentityState {
+  signedIn: boolean;
+  accountId: string;
+  disabledAccountId: string;
+  accessToken: string;
+  permissions: number;
+}
+
+export const IdentityContext = createContext<IdentityState>({
+  signedIn: false,
+  accountId: '',
+  disabledAccountId: '',
+  accessToken: '',
+  permissions: 0,
+});
+
+export const createLegacyIdentityContext = (state: InitialState) => {
+  return {
+    signedIn: !!state.meta.me,
+    accountId: state.meta.me,
+    disabledAccountId: state.meta.disabled_account_id,
+    accessToken: state.meta.access_token,
+    permissions: state.role?.permissions ?? 0,
+  };
+};
+
+export const useIdentity = (): IdentityState => {
```

Comment:
I dont think you need `: IdentityState` here, Typescript should infer it correctly?